features/steps/deciders.py

```diff
@@ -0,0 +1,64 @@
+class StepDecider:
+    """
+    Determines at the step level if a test should be run.
+    """
+
+    def __init__(self, context):
+        """
+        Behave context of currently running step.
+        :param context: A Behave framework context
+        """
+        self.context = context
+
+    def __skip_context(self):
+        return self.context.vendor_skip
+
+    def should_run_test(self):
+        return not self.__skip_context()
+
+
+class ResourceDecider:
+    """
+    Determines if a resource should be validated based on Argonaut Value Sets.
+    """
+    argo_vital_codes = ["8716-3",
+                        "9279-1",
+                        "8867-4",
+                        "59408-5",
+                        "8310-5",
+                        "8302-2",
+                        "8306-3",
+                        "8287-5",
+                        "29463-7",
+                        "39156-5",
+                        "55284-4",
+                        "8480-6",
+                        "8462-4"]
+
+    def __init__(self, resource):
+        self.resource = resource
+
+    def __needs_argo_validation(self):
+
+        validate = False
+        found_validation_toggle = False
+
+        try:
```

Comment:
The logic here looks right, but it's a bit hard to follow. I wonder if a "guard" approach might be clearer, like the following pseudocode

```
if resource type != Observation: return True
if category != vitals: return True
if code in "our vitals loincs": return True

return False
```

Reply:
Once we need more logic here, it would make sense to call to a resource-type-specific "should validate" functions, rather than packing all of the logic in here. But that's generalization for later, I think...